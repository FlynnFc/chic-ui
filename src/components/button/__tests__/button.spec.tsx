import Button from '..';
import { render } from '../../../test/utils';
import SearchIcon from '../../../icons/Search';

/* eslint-disable */
describe('Button Component', () => {
  it('Renders Correctly', () => {
    const ButtonGroup = render(
      <>
        <div>
          <Button size="large">Large</Button>
          <Button size="default">Default</Button>
          <Button size="small">Small</Button>
        </div>
        <div>
          <Button type="primary">Primary</Button>
          <Button type="secondary">Secondary</Button>
          <Button type="warning">Warning</Button>
          <Button type="success">Success</Button>
          <Button type="danger">Danger</Button>
          <Button type="info">Info</Button>
          <Button type="light">Light</Button>
        </div>
        <div>
          <Button loading type="secondary" />
          <Button loading type="light" />
          <Button disabled>Disabled</Button>
          <Button type="light" icon={SearchIcon}>
            Search
          </Button>
        </div>
      </>
    );
    expect(ButtonGroup).toMatchSnapshot();
  });
});
