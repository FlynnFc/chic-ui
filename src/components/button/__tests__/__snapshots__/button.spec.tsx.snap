// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component Renders Correctly 1`] = `
Array [
  <div>
    <button
      className="sc-bdfBQB iuzAqj"
      disabled={false}
      size="large"
      type="button"
    >
      Large
    </button>
    <button
      className="sc-bdfBQB jOgVcX"
      disabled={false}
      size="default"
      type="button"
    >
      Default
    </button>
    <button
      className="sc-bdfBQB spVYT"
      disabled={false}
      size="small"
      type="button"
    >
      Small
    </button>
  </div>,
  <div>
    <button
      className="sc-bdfBQB jOgVcX"
      disabled={false}
      size="default"
      type="button"
    >
      Primary
    </button>
    <button
      className="sc-bdfBQB ifOkXl"
      disabled={false}
      size="default"
      type="button"
    >
      Secondary
    </button>
    <button
      className="sc-bdfBQB dGipFf"
      disabled={false}
      size="default"
      type="button"
    >
      Warning
    </button>
    <button
      className="sc-bdfBQB gppRuv"
      disabled={false}
      size="default"
      type="button"
    >
      Success
    </button>
    <button
      className="sc-bdfBQB hKkpIX"
      disabled={false}
      size="default"
      type="button"
    >
      Danger
    </button>
    <button
      className="sc-bdfBQB hZGcwn"
      disabled={false}
      size="default"
      type="button"
    >
      Info
    </button>
    <button
      className="sc-bdfBQB bCHUX"
      disabled={false}
      size="default"
      type="button"
    >
      Light
    </button>
  </div>,
  <div>
    <button
      className="sc-bdfBQB lcrjrz"
      disabled={false}
      size="default"
      type="button"
    >
      Loading
      <span
        className="sc-dlfnuX debHjb"
        size={20}
      />
    </button>
    <button
      className="sc-bdfBQB iHGYdF"
      disabled={false}
      size="default"
      type="button"
    >
      Loading
      <span
        className="sc-dlfnuX debHjb"
        size={20}
      />
    </button>
    <button
      className="sc-bdfBQB FnpUW"
      disabled={true}
      size="default"
      type="button"
    >
      Disabled
    </button>
    <button
      className="sc-bdfBQB bCHUX"
      disabled={false}
      size="default"
      type="button"
    >
      Search
      <svg
        className="sc-gsTEea cDZHpU"
        viewBox="0 0 487.95 487.95"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M481.8 453l-140-140.1c27.6-33.1 44.2-75.4 44.2-121.6C386 85.9 299.5.2 193.1.2S0 86 0 191.4s86.5 191.1 192.9 191.1c45.2 0 86.8-15.5 119.8-41.4l140.5 140.5c8.2 8.2 20.4 8.2 28.6 0 8.2-8.2 8.2-20.4 0-28.6zM41 191.4c0-82.8 68.2-150.1 151.9-150.1s151.9 67.3 151.9 150.1-68.2 150.1-151.9 150.1S41 274.1 41 191.4z"
        />
      </svg>
    </button>
  </div>,
]
`;
